Assets/Scripts/Spawner/ResourceReg.cs

```diff
@@ -0,0 +1,106 @@
+﻿using Ecosystem.Grid;
+using System.Collections.Generic;
+using Unity.Collections;
+using UnityEngine;
+
+
+
+
+/// <summary>
+/// Spawns the prefab given as seaweed and the prefab given as grass in the water and on land respectively.
+/// Spwans happens at predecided intervals 
+/// </summary>
+public class ResourceReg : MonoBehaviour
+{
+
+    public GameObject grass;
+    public float GrassSpawnTime;
+    private float deltaTimeGrass = 0f;
+    private float searchTime = 0f;
+
+    private bool lookingForFreeTile = true;
+    private int gridSize;
+
+    public GameObject seaweed;
+    public float SeaweedSpawnTime;
+    private float deltaTimeSeaweed = 0f;
+
+    public static List<int> occupiedWaterSpace = new List<int>();
```

Comment:
```suggestion
    private List<int> occupiedWaterSpace = new List<int>();
```
Ska den vara public static?

Reply:
Tanken var att göra den public static för att kunna ta bort tilens nummer från listan när sjögräset blir uppätet, då skulle problemet att saker inte kan spawna där andra saker har spawnat innan lösas. Är detta en okej lösning?

Reply:
Jag tycker inte det. Det här problemet ligger lite utanför scopet av den här regeneration-featuren. Vi kanske får diskutera det i gruppen.

Om vi vill kunna se om en ruta är upptagen så borde det skötas utanför det här systemet och fungera för alla typer av object som kan placeras i världen. Man skulle kunna ha en array med alla placerade object (träd, gräs, buskar, etc.), både walkable och unwalkable. En annan lösning skulle kunna vara att använda colliders för att inte spawna saker i varandra (typ en collider check där man tänkt spawna).